### PR TITLE
Added HDF documentation.

### PR DESCRIPTION
**Pull request by airvzxf**
### :pencil: Description

**Type:** :memo: `documentation`

Write a complete description of your changes, including the necessary context or any piece of information required to understand your work.

> Added the HDF documentation with two main sections: the overview and the extraction of the data in the H5 files.

Also, link issues affected by this pull request by using the keywords: `close`, `closes`, `closed`, `fix`, `fixes`, `fixed`, `resolve`, `resolves` or `resolved`. 

### :pushpin: Resources

Examples, notebooks, and links to useful references.

Screenshot of my local server

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [x] Other method (describe): Manual, I build the docs and run on the local server.
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [x] I updated the documentation according to my changes
- [x] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.
@wkerzendorf 
@andrewfullard
@atharva-2001

### CODE REVIEW

**Comment by atharva-2001** [comment]
I'm afraid I can't see the documentation on the website sent by the bot. Is the bot not working correctly?

**Comment by KasukabeDefenceForce** [comment]
> I'm afraid I can't see the documentation on the website sent by the bot. Is the bot not working correctly? @KasukabeDefenceForce

The issue is here, when the LFS setup step is run it checks out the main repo, which is why the documentation generated is based upon the files in the master.
https://github.com/tardis-sn/tardis/actions/runs/9266497498/job/25490895572?pr=2630#step:5:15

File: docs/io/hdf/index.rst [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+
+Extract data
+============
+
+
+Extract data using the HDF5 libraries
+-------------------------------------
+
+In TARDIS the generated data is valuable because it has been reviewed and validated. For this reason, TARDIS usually takes the generated data.
```
Comment (andrewfullard) [inline docs/io/hdf/index.rst:59]: `For this reason, TARDIS tests against the generated data.`

Comment (airvzxf) [reply docs/io/hdf/index.rst]: Done in the commit: 0843a558d89975fbe48d3e0e17654a3dc57bf08d.

**Comment by jvshields** [comment]
This looks really good to me! My one comment is that I'm not sure the "Extract Data" section really belongs here, or does a great job explaining how to use hdfs in tardis. What's currently there seems more specific to our regression testing, but I don't think somebody would go to the HDF section under I/O to try to find this type of information, and I think a better thing to have here would just be an example of reading an hdf and showing its contents.

**Comment by airvzxf** [comment]
> This looks really good to me! My one comment is that I'm not sure the "Extract Data" section really belongs here, or does a great job explaining how to use hdfs in tardis. What's currently there seems more specific to our regression testing, but I don't think somebody would go to the HDF section under I/O to try to find this type of information, and I think a better thing to have here would just be an example of reading an hdf and showing its contents.

I could split this comment into two different parts. One is the location for this information and the other for the content of this section.

Talking about the location, this is the general structure:
```text
Installation
Quickstart for TARDIS
Tutorials
How-To Guides
Frequently Asked Questions
API

Input/Output
    Configuration (Required Input)
    Reading Models
    Optional Inputs
    Visualization Tools & Widgets
    Additional Outputs

Physics Walkthrough
    Physics Walkthrough Introduction
    Setting Up the Simulation
    Monte Carlo Iteration
    Updating Plasma and Convergence
    Spectrum Generation

Contributing to TARDIS
    Contribution Guidelines
    Developer Workflow
    Developer Tools
    Change Log
    Features In-Progress

Other Resources
    Credits & Publication Policies
    Papers Using TARDIS
    Code Comparison
    References and Glossary
```

What could be the best place to insert it?

**Review by andrewfullard — APPROVED** [review]
I think it is fine to get this merged and move it around once it's in.

**Review by jvshields — APPROVED** [review]
Andrew makes a reasonable point.

**Pull request merged**